helix-core/src/surround.rs

```diff
@@ -52,7 +57,58 @@ pub fn get_pair(ch: char) -> (char, char) {
         .unwrap_or((ch, ch))
 }
 
+/// Find the position of surround pairs of any [`PAIRS`] using tree-sitter when possible.
+///
+/// Returns a tuple `(anchor, head)`, meaning it is not always ordered.
 pub fn find_nth_closest_pairs_pos(
+    syntax: Option<&Syntax>,
+    text: RopeSlice,
+    range: Range,
+    skip: usize,
+) -> Result<(usize, usize)> {
+    match syntax {
+        Some(syntax) => find_nth_closest_pairs_ts(syntax, text, range, skip),
+        None => find_nth_closest_pairs_plain(text, range, skip),
+    }
+}
+
+fn find_nth_closest_pairs_ts(
+    syntax: &Syntax,
+    text: RopeSlice,
+    range: Range,
+    skip: usize,
+) -> Result<(usize, usize)> {
+    let is_close_pair = |ch| PAIRS.iter().any(|(_, close)| *close == ch);
```

Comment:
I think autopairs could stay separate for now.

For example.ple we don't want to add `'` there since that I'd also often used as standalone (like rust lifetimes).

Although I guess that could also be handled by language co fig (autpais are co figural iirc). Thoughts @the-mikedavis

Reply:
Yeah let's keep auto-pairs separate for now, those need some special logic for things like `'` as Pascal says

Reply:
Done in 938222478301233fd7328ec2aa59a33381e35060
No new functionality added, only moved PAIRS-related stuff to `match_bracking.rs`. I think it's a better place than `surround.rs` or `helix_core::lib.rs`.